src/main/java/com/jhuguet/sb_taskv1/app/services/impl/UserServiceImpl.java

```diff
@@ -117,40 +116,62 @@ private Tag countTagIterations(Order order) throws IdNotFound, NoTagInOrder {
 
     private Map<Integer, Integer> mapCurrentTags() {
         Map<Integer, Integer> currentTags = new HashMap<>();
-        tagRepository
-                .findAll()
-                .forEach(t -> currentTags.put(t.getId(), 0));
+        tagRepository.findAll().forEach(t -> currentTags.put(t.getId(), 0));
         return currentTags;
     }
 
     private Integer getMaxTagID(Map<Integer, Integer> tags) {
-        return tags
-                .entrySet()
-                .stream()
-                .max(Map.Entry.comparingByValue())
-                .get()
-                .getValue();
+        return tags.entrySet().stream().max(Map.Entry.comparingByValue()).get().getValue();
     }
 
     @Override
-    public User signIn(User user) throws MissingEntity, MissingRequiredFields {
+    public void signIn(User user) throws MissingEntity, MissingRequiredFields, UnqualifiedAuthority {
         if (!Objects.isNull(user)) {
             userEntityValidations(user);
-            userRepository.save(user);
+            userRepository.save(assembleUserToSave(user));
         } else {
             throw new MissingEntity();
         }
-        return null;
     }
 
-    private void userEntityValidations(User user) throws MissingRequiredFields {
-        if (user
-                .getPassword()
-                .equals("") || user
-                .getUsername()
-                .equals("")) {
+    private void userEntityValidations(User user) throws MissingRequiredFields, UnqualifiedAuthority {
+        if (user.getPassword().equals("") || user.getUsername().equals("")) {
             throw new MissingRequiredFields();
         }
+
+        if (user.getUsername().equalsIgnoreCase("administrator")) {
+            throw new UnqualifiedAuthority();
+        }
+    }
+
+    private User assembleUserToSave(User user) {
+        return new User(user.getUsername(), user.getEmail(), passwordEncoder.encode(user.getPassword()),
+                new HashSet<>(), "USER");
+    }
+
+    public void matchPasswords(String username, String password) throws WrongCredentials {
+        User user = userRepository.findByUsername(username);
```

Comment:
I suggest you to check matching username and password by extending AuthenticationPrivider. Check it https://www.springcloud.io/post/2022-05/spring-security-badcredentialsexception/#gsc.tab=0

Reply:
But this should be basically using the DAOAuthenticationProvider. Should I retrace what I have and change it to the DAO one? Because the logic is there, but we don't actually see what is the user and password matching that is being done